engine/core/src/main/java/org/datacleaner/job/builder/AnalysisJobBuilder.java

```diff
@@ -774,9 +785,11 @@ public AnalysisJobMetadata createMetadata() {
      * 
      * @return
      * @throws IllegalStateException
-     *             if the job is invalidly configured.
+     *             if the job is invalidly configured. See
+     *             {@link #isConfigured(boolean)} for detailed exception
+     *             descriptions.
      */
-    public AnalysisJob toAnalysisJob() throws IllegalStateException {
+    public AnalysisJob toAnalysisJob() throws RuntimeException {
```

Comment:
The doc needs an update yes. But do you think there's anything wrong with both having it in javadoc and as a method modifier? IMO that is best practice?